Firestore/core/src/firebase/firestore/core/query.cc

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2018 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "Firestore/core/src/firebase/firestore/core/query.h"
+
+#include "Firestore/core/src/firebase/firestore/model/document_key.h"
+#include "Firestore/core/src/firebase/firestore/model/resource_path.h"
+
+namespace firebase {
+namespace firestore {
+namespace core {
+
+using model::Document;
```

Comment:
I'm interpreting this as a 'sibling namespace':

"What if a name lives in a sibling namespace, such as ::example::tools::Thing? You can say either tools::Thing or ::example::tools::Thing. The fully qualified name is always correct, but it may also be appropriate to use the relative name. Use your own judgment."

That said, I'm happy to change it.

You know what? I'll throw together a short doc outlining our options. We can quickly vote on what we want to do going forward, and then just cleanup as we go.

Reply:
Note that we're already using the sibling namespace thing to cut down on verbosity in the header. If we can't count on it to work here how could we count on it to work there?

(And to be clear: I'm not arguing to make everything in the headers fully qualified too.)